### PR TITLE
Workaround to accept empty tag values

### PR DESCRIPTION
**Pull request by prateek**

### CODE REVIEW

File: serialize/decoder.go [hunk 1]
```diff
@@ -111,12 +111,18 @@ func (d *decoder) decodeTag() (ident.Tag, error) {
 	if err != nil {
 		return ident.Tag{}, err
 	}
+	if len(name.Bytes()) == 0 {
+		return ident.Tag{}, errEmptyTagNameLiteral
```
Comment (robskillington) [inline serialize/decoder.go]: Can you wrap this in an invalid params error (to return Bad Request)?  `xerrors.NewInvalidParamsError(..)`?

File: serialize/encoder.go [hunk 2]
```diff
@@ -52,12 +52,17 @@ import (
 var (
 	byteOrder        binary.ByteOrder = binary.LittleEndian
 	headerMagicBytes                  = encodeUInt16(headerMagicNumber)
+
+	// FOLLOWUP(prateek): this terrible hack is necessary due to a bug in the indexing
+	// sub-system. Our FST library (Vellum) doesn't handle '' values correctly in an edge
+	// case. I'll remove this once I have some cycles to come back to it.
+	mapEmptyTagValueToSpaceHack = ident.BytesID([]byte(" "))
```
Comment (robskillington) [inline serialize/encoder.go]: Can you wrap this in an invalid params error (to return Bad Request)?  `xerrors.NewInvalidParamsError(..)`?

File: serialize/encoder.go [hunk 3]
```diff
@@ -52,12 +52,17 @@ import (
...
 )
 
 var (
-	errTagEncoderInUse   = errors.New("encoder already in use")
-	errTagLiteralTooLong = errors.New("literal is too long")
-	errEmptyTagLiteral   = errors.New("tag nam/value cannot be empty")
+	errTagEncoderInUse     = errors.New("encoder already in use")
+	errTagLiteralTooLong   = errors.New("literal is too long")
+	errEmptyTagNameLiteral = errors.New("tag name cannot be empty")
```
Comment (robskillington) [inline serialize/encoder.go]: Can you wrap this in an invalid params error (to return Bad Request)?  `xerrors.NewInvalidParamsError(..)`?

Comment (prateek) [reply serialize/encoder.go]: sure thing

File: client/errors.go [hunk 4]
```diff
@@ -46,6 +46,9 @@ func IsBadRequestError(err error) bool {
 		if e, ok := err.(*rpc.Error); ok && tterrors.IsBadRequestError(e) {
 			return true
 		}
+		if e := xerrors.GetInnerInvalidParamsError(err); e != nil {
```
Comment (robskillington) [inline client/errors.go]: Nice.

**Pull request merged**